app/v5/createForeignPropertyPeriodCumulativeSummary/def1/Def1_CreateForeignPropertyPeriodCumulativeSummaryRulesValidator.scala

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v5.createForeignPropertyPeriodCumulativeSummary.def1
+
+import api.controllers.validators.RulesValidator
+import api.controllers.validators.resolvers.{ResolveFromAndToDates, ResolveParsedCountryCode, ResolveParsedNumber}
+import api.models.errors._
+import cats.data.Validated
+import cats.data.Validated.Invalid
+import cats.implicits.toTraverseOps
+import v5.createForeignPropertyPeriodCumulativeSummary.def1.model.request.{
+  Def1_CreateForeignPropertyPeriodCumulativeSummaryRequestData,
+  Expenses,
+  ForeignProperty
+}
+
+object Def1_CreateForeignPropertyPeriodCumulativeSummaryRulesValidator
+    extends RulesValidator[Def1_CreateForeignPropertyPeriodCumulativeSummaryRequestData] {
+
+  private val resolveParsedNumber = ResolveParsedNumber()
+
+  def validateBusinessRules(parsed: Def1_CreateForeignPropertyPeriodCumulativeSummaryRequestData)
+      : Validated[Seq[MtdError], Def1_CreateForeignPropertyPeriodCumulativeSummaryRequestData] = {
+    import parsed.body._
+
+    combine(
+      ResolveFromAndToDates.withOptions((fromDate, toDate)).getOrElse(valid),
+      validateForeignProperty(foreignProperty)
+    ).onSuccess(parsed)
+  }
+
+  private def validateForeignProperty(foreignProperty: Seq[ForeignProperty]): Validated[Seq[MtdError], Unit] = {
+    val zippedForeignProperties = foreignProperty.zipWithIndex
+
+    val validatedCountryCodes = zippedForeignProperties
+      .map { case (entry, index) =>
+        (entry.countryCode, s"/foreignProperty/$index/countryCode")
+      }
+      .groupBy(_._1)
+      .collect {
+        case (code, codeAndPaths) if codeAndPaths.size >= 2 =>
+          Invalid(List(RuleDuplicateCountryCodeError.forDuplicatedCodesAndPaths(code, codeAndPaths.map(_._2))))
```

Comment:
We should not be doing this validation (should be done downstream instead). Need to discuss what other errors we should not be validating?